### DISCUSSION TITLE
how do i run npm install --legacy-peer-deps for some particular repo/project

### DISCUSSION THREAD

**Discussion by merajshaikh** [renovatebot/renovate#22499] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.104-slim

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Hi,
I am running renovate for my whole Gitlab but there are some projects with legacy stuff
so they need --legacy-peer-deps while installing packages
how can I specify which projects to pick up and do this?

### What you tried so far.

i tried this does not seem to work

{
      "packagePatterns": [
        "my-project",< this is from package.json
      ],
      "depType": "peer"
    }

### Relevant debug logs

<details><summary>Logs</summary>

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: semantic-release-ms-teams@2.1.0
npm ERR! Found: semantic-release@21.0.1
npm ERR! node_modules/semantic-release
npm ERR!   semantic-release@"21.0.1" from the root project
npm ERR!   peer semantic-release@">=18.0.0" from @semantic-release/changelog@6.0.3
npm ERR!   node_modules/@semantic-release/changelog
npm ERR!     @semantic-release/changelog@"6.0.3" from the root project
npm ERR!   6 more (@semantic-release/commit-analyzer, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer semantic-release@"19" from semantic-release-ms-teams@2.1.0
npm ERR! node_modules/semantic-release-ms-teams
npm ERR!   semantic-release-ms-teams@"2.1.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: semantic-release@19.0.5
npm ERR! node_modules/semantic-release
npm ERR!   peer semantic-release@"19" from semantic-release-ms-teams@2.1.0
npm ERR!   node_modules/semantic-release-ms-teams
npm ERR!     semantic-release-ms-teams@"2.1.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

```

</details>

**Comment by rarkins** [comment]
Projects which require legacy peer deps should ideally have a `.npmrc` checked into the repository indicating as such.

If you can't do that, then you can configure `config.npmrc` in `renovate.json` in the repos.

Be sure to configure `npmrcMerge=true` if you need either of those approaches to be combined with a "global" npmrc

**Reply by merajshaikh** [reply]
hi @rarkins 
can you please give me 1 example of how I can do it from renovate.json
I already have npmrcmerge true enabled for other stuff

**Reply by rarkins** [reply, marked as answer]
`"npmrc": "legacy-peer-deps=true"`

**Reply by merajshaikh** [reply]
Thank you so much it worked @rarkins
